tests/cnf/core/network/security/tests/nftables-custom-firewall-test.go

```diff
@@ -0,0 +1,505 @@
+	Expect(len(podList)).ToNot(BeZero(), "Pod list is empty")
+	Expect(len(nextHopList)).ToNot(BeZero(), "Next hop IP addresses list is empty")
+	Expect(len(nextHopList)).To(BeNumerically(">=", len(podList)),
+		fmt.Sprintf("Number of speaker IP addresses[%d] is less than the number of pods[%d]",
+			len(nextHopList), len(podList)))
+
+	routesMap := make(map[string]string)
+
+	for _, frrPod := range podList {
+		if frrPod.Definition.Spec.NodeName == workerNodeList[0].Definition.Name {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[1]
+		} else {
+			routesMap[frrPod.Definition.Spec.NodeName] = nextHopList[0]
+		}
+	}
+
+	return routesMap
+}
+
+func setStaticRoute(frrPod *pod.Builder, action, destIP string, nextHopMap map[string]string) (string, error) {
+	buffer, err := frrPod.ExecCommand(
+		[]string{"ip", "route", action, destIP, "via", nextHopMap[frrPod.Definition.Spec.NodeName]})
+	if err != nil {
+		if strings.Contains(buffer.String(), "File exists") {
+			glog.V(90).Infof("Warning: Route to %s already exist", destIP)
+
+			return buffer.String(), nil
+		}
+
+		if strings.Contains(buffer.String(), "No such process") {
+			glog.V(90).Infof("Warning: Route to %s already absent", destIP)
+
+			return buffer.String(), nil
+		}
+
+		return buffer.String(), err
+	}
+
+	return buffer.String(), nil
+}
+
+// DefineBGPConfigMapWithStaticRouteAndNetwork defines BGP config file with static route and network.
+func defineConfigMapWithStaticRouteAndNetwork(hubPodIPs, nodeIPAddresses []string) string {
+	bgpConfig :=
+		fmt.Sprintf("ip route %s/32 %s\n", nodeIPAddresses[1], hubPodIPs[0]) +
+			fmt.Sprintf("ip route %s/32 %s\n!\n", nodeIPAddresses[0], hubPodIPs[1])
+
+	bgpConfig += "!\nline vty\n!\nend\n"
+
+	return bgpConfig
+}
+
+func createTestPodsOnWorkers(nodeName string, portNum int) *pod.Builder {
+	testPod, err := pod.NewBuilder(
+		APIClient, "testpod-"+nodeName, tsparams.TestNamespaceName, NetConfig.CnfNetTestContainer).
+		DefineOnNode(nodeName).WithHostNetwork().WithHostPid(true).
+		RedefineDefaultCMD([]string{"/bin/bash", "-c",
+			fmt.Sprintf("testcmd -interface br-ex -protocol tcp -port %d -listen", portNum)}).
+		WithPrivilegedFlag().CreateAndWaitUntilRunning(180 * time.Second)
+	Expect(err).ToNot(HaveOccurred(), "Failed to create nginx test pod")
+
+	return testPod
+}
+
+func validateTCPTraffic(clientPod *pod.Builder, destIPAddrs []string, interfaceName,
+	containerName string) error {
+	for _, destIPAddr := range removePrefixFromIPList(destIPAddrs) {
+		command := fmt.Sprintf("testcmd -interface %s -protocol tcp -port %d -server %s", interfaceName,
+			8888, destIPAddr)
+		_, err := clientPod.ExecCommand([]string{"bash", "-c", command}, containerName)
+
+		return err
+	}
+
+	return nil
+}
+
+func createMCAndWaitforMCPStable(fileContentString string) {
+	truePointer := true
+	stringgzip := "gzip"
+
+	mode := 384
+	fileContents := fileContentString
```

Comment:
?

Reply:
that is the base64 string

Reply:
what does it mean? 
you can use it
`Source:      &fileContentString,`

Reply:
I am using mc WithRawConfig to create and delete the machineConfig. This is using IgnitionConfig. So I am using the fileContents and not directly adding to Source: